OsmAnd/src/net/osmand/plus/views/layers/NearbyPlacesLayer.java

```diff
@@ -240,6 +222,46 @@ public PointDescription getObjectName(Object o) {
 	@Override
 	public void collectObjectsFromPoint(PointF point, RotatedTileBox tileBox, List<Object> res,
 	                                    boolean unknownLocation, boolean excludeUntouchableObjects) {
+		List<NearbyPlacePoint> points = customObjectsDelegate.getMapObjects();
+		if (!points.isEmpty()) {
+			getNearbyPLaceFromPoint(tileBox, point, res, points);
+		}
+	}
+
+	private void getNearbyPLaceFromPoint(RotatedTileBox tb, PointF point, List<? super NearbyPlacePoint> res, List<NearbyPlacePoint> points) {
+		MapRendererView mapRenderer = getMapRenderer();
+		float radius = getScaledTouchRadius(getApplication(), tb.getDefaultRadiusPoi()) * TOUCH_RADIUS_MULTIPLIER;
+		List<PointI> touchPolygon31 = null;
+		if (mapRenderer != null) {
+			touchPolygon31 = NativeUtilities.getPolygon31FromPixelAndRadius(mapRenderer, point, radius);
+			if (touchPolygon31 == null) {
+				return;
+			}
+		}
+
+		for (NearbyPlacePoint nearbyPoint : points) {
+			double lat = nearbyPoint.getLatitude();
+			double lon = nearbyPoint.getLongitude();
+			boolean add = mapRenderer != null
+					? NativeUtilities.isPointInsidePolygon(lat, lon, touchPolygon31)
+					: tb.isLatLonNearPixel(lat, lon, point.x, point.y, radius);
+			if (add) {
+				res.add(nearbyPoint);
+			}
+		}
+	}
+
+	@Nullable
+	private Object getSelectedObject() {
+		MapActivity mapActivity = getMapActivity();
+		if (mapActivity != null) {
+			MapContextMenu mapContextMenu = mapActivity.getContextMenu();
+			Object object = mapContextMenu.getObject();
+			if (mapContextMenu.isVisible()) {
+				return object;
```

Comment:
Should selectedObject be only NearbyPlacePoint? we get all types of objects here

Reply:
Do what @Chumva suggested
Looks like it should return `NearbyPlacePoint` and not `Object`